### ISSUE TITLE
Covsirphy handing of ordinal atriubtes.

### ISSUE THREAD

**Issue by baluballa** [lisphilar/covid19-sir#965]
### Covsirphy handing of ordinal atriubtes.

I have a complete set oxcgrt_data from the policy tracker (https://github.com/OxCGRT/covid-policy-tracker). Some of these data is numbered with respect to measurement. eg , the H6_Facial Coverings has an `ordinal scale of`:
```
0 - No policy
1 - Recommended
2 - Required in some specified shared/public spaces outside the home with other people present, or some situations when social distancing not possible
3 - Required in all shared/public spaces outside the home with other people present or all situations when social distancing not possible
4 - Required outside the home at all times regardless of location or presence of other people
```
(https://github.com/OxCGRT/covid-policy-tracker/blob/master/documentation/codebook.md)

In covsirphy's documentations example it seems covsirphy handles this by removing cols like: 
```python
removed_cols=["Stringency_index", "Tests", "Tests_diff", "Vaccinations"]
```
However i would like to measure effect of the policy "1 recomended". So question is, is this possible?

```Python
import covsirphy as cs
```

**Comment by lisphilar** [comment]
I'm not sure if this is the answer you're looking for, but the values (0, 1, 2, 3, 4) of "H6_Facial Converings" can be registered with `extras` keyword.
```Python
import covsirphy as cs
loader = cs.DataLoader()
jhu_data = loader.jhu()
oxcgrt_data = loader.oxcgrt()
snl = cs.Scenario(country="Norway")
snl.register(jhu_data, extras=[oxcgrt_data])
snl.records(variables=["Contact_tracing"])
```

**Comment by baluballa** [comment]
My `ocxcgrt_data` looks like this 
```python
og_x_data = ["C1_School closing", "C2_Workplace closing", "C3_Cancel public events", 
"C4_Restrictions on gatherings", "C5_Close public transport", "C6_Stay at home requirements", 
"C7_Restrictions on internal movement", "C8_International travel controls", "E1_Income support", 
"E2_Debt/contract relief", "E3_Fiscal measures", "E4_International support", "H1_Public information campaigns", 
"H2_Testing policy", "H3_Contact tracing", "H4_Emergency investment in healthcare", "H5_Investment in vaccines", 
"H6_Facial Coverings", "H7_Vaccination policy", "H8_Protection of elderly people", "M1_Wildcard, ConfirmedCases", 
"ConfirmedDeaths", "StringencyIndex", "StringencyIndexForDisplay", "StringencyLegacyIndex", "StringencyLegacyIndexForDisplay", 
"GovernmentResponseIndex", "GovernmentResponseIndexForDisplay", "ContainmentHealthIndex", "ContainmentHealthIndexForDisplay", 
"EconomicSupportIndex", "EconomicSupportIndexForDisplay"]

data_loader.lock(
   ...
   oxcgrt_variables=og_x_data,
   ...
```
Where the data in list is from csv from the policy trackers github and the linst contains columns headers.

Ordinal data means the categorical data order has a meaning. So if want to check impact on only reccomend facial covers I look for where  `H6_Facial Coverings <= # Required `

**Comment by lisphilar** [comment]
Do you mean you want to create a new column `face_covering_required` (1 when required, 0 when not required) to the OxCGRT data?

**Comment by baluballa** [comment]
In some cases the order of data means something, eg: `Best, Good, Neutral, Bad, Worse.` rather than eg, Colors, "Blue, Green, Red, Yellow" and so on... So if I want to test who has an required resttricition in facial coverings, I will check for values above 1. Recommended, cause above that values more constraint.

Its hard for me to explain, so found an article at this site (https://stats.idre.ucla.edu/other/mult-pkg/whatstat/what-is-the-difference-between-categorical-ordinal-and-interval-variables/) which explains it.

**Comment by lisphilar** [comment]
Yes ordinal data has meaning in the order of items.

I'm sorry, but I didn't see the point of your question. How to anlyse the impact of facial coverings (required or not
) on ODE parameters and so on?

**Comment by baluballa** [comment]
That is the idea. If you have simulated data, you can train data with the oxcgrt data. I am just not sure how ordinal data data is handled, when "fitted". 

Usually in Machine Learning this is handled in the _data preparation phase_. I guess I could generate future data with setting I want to measure eg ` H6_Facial ` = 3 and then simualted with this setting injected into the _oxcgrt_variables_ someway. Just wonder if someone has tested this? Again not a big issue, I can investigate myself.

Regards

Thor

**Comment by lisphilar** [comment]
Ordinal data (encoded with label encoder) is regarded as numerical data at the latest version. As you said (if I get your point), ordinal data should be ended with One Hot Encoder in the _data preparation phase_.

**Comment by lisphilar** [comment]
Is it possible to calculate the scores of the categories with agent-based models #396?

**Comment by lisphilar** [comment]
Discuss this point later, if necessary.
